### PR TITLE
Userspace and keymap update for user bbaserdem.

### PR DESCRIPTION
**Pull request by bbaserdem**
## Description

I completely refactored my code; and cleaned up my userspace. The following were done;
* Migrate all keyboards that can be migrated to the layouts/community folder
* Fixed functionality that was broken due to changes in qmk code since last accepted pull request.
* Included additional functionality.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python); tried to be consistent in styling; but took liberties to make the code readable; and might have slipped up here and there with styling.
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: layouts/community/split_3x6_3/bbaserdem/halconf.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (drashna) [inline layouts/community/split_3x6_3/bbaserdem/halconf.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: layouts/community/split_3x6_3/bbaserdem/mcuconf.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (drashna) [inline layouts/community/split_3x6_3/bbaserdem/mcuconf.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: layouts/community/split_3x6_3/bbaserdem/config.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+#   define SPLIT_WPM_ENABLE
+
+    // Audio output on pin A5 (actualy will be output on B1 using keymap init)
+#   ifdef AUDIO_ENABLE
+#       define AUDIO_PIN A5
+#       define MUSIC_MAP
+#   endif
+
+    // Communication has to be serial on ARM
+#   define USE_SERIAL_PD2
```
Comment (drashna) [inline layouts/community/split_3x6_3/bbaserdem/config.h:48]: This define doesn't actually do anything.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: layouts/community/split_3x5_3/bbaserdem/config.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#pragma once
+
+// Split definition; use the following to flash;
+//  pro-micro:      avrdude-split-left/right
+//  elite-c:        dfu-split-left/right
+#define EE_HANDS
+
+#ifdef KEYBOARD_crkbd_rev1
+    // Corne overrides
+    #include "../../../../keyboards/crkbd/rev1/config.h"
```
Comment (drashna) [inline layouts/community/split_3x5_3/bbaserdem/config.h:27]: This should not be needed. 
```suggestion
```

File: layouts/community/split_3x6_3/bbaserdem/chconf.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+// Code pertaining the ARM crkbd version
+#ifdef KEYBOARD_crkbd_rev1
+    #define CH_CFG_ST_RESOLUTION 16
+    #define CH_CFG_ST_FREQUENCY 10000
+    #if __has_include("platforms/chibios/common/configs/chconf.h")
+        #    include_next "platforms/chibios/common/configs/chconf.h"
+    #else
+        #    include_next "chconf.h"
+    #endif
```
Comment (drashna) [inline layouts/community/split_3x6_3/bbaserdem/chconf.h:26]: Should only need this, now.
```suggestion
#    include_next "chconf.h"
```

File: layouts/community/split_3x6_3/bbaserdem/config.h [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+// Split definition; use the following to flash;
+//  pro-micro:      avrdude-split-left/right
+//  elite-c:        dfu-split-left/right
+
+// Code pertaining to Corne ARM; from Waffle's keymap
+#ifdef KEYBOARD_crkbd_rev1
+#   include "config_common.h"
```
Comment (drashna) [inline layouts/community/split_3x6_3/bbaserdem/config.h:24]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/splitkb/kyria/keymaps/bbaserdem/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+This file contains my Kyria board layout.
```
Comment (fauxpark) [inline keyboards/splitkb/kyria/keymaps/bbaserdem/README.md:1]: Please rename this file to lowercase.

File: layouts/community/ortho_3x10/bbaserdem/README.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+This file contains my [3x10 ortho](../../../default/ortho_3x10) board layouts
```
Comment (fauxpark) [inline layouts/community/ortho_3x10/bbaserdem/README.md:1]: As above.

File: layouts/community/ortho_4x12/bbaserdem/README.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+This file contains my [4x12 ortho](../../../default/ortho_4x12) board layouts
```
Comment (fauxpark) [inline layouts/community/ortho_4x12/bbaserdem/README.md:1]: ""

File: users/bbaserdem/rules.mk [hunk 10]
```diff
@@ -1,21 +1,91 @@
...
+BOOTMAGIC_ENABLE = no   # Access to EEPROM settings, not needed
+CONSOLE_ENABLE = no     # Allows console output with a command
+COMMAND_ENABLE = no     # Some bootmagic thing i dont use
+SLEEP_LED_ENABLE = no   # Breathes LED's when computer is asleep. Untested.
+MIDI_ENABLE = no        # Midi driver (untested)
+UNICODE_ENABLE = no 	# We use unicodemap, not unicode
+UCIS_ENABLE = no 	# We use unicodemap, not ucis
+FAUXCLICKY_ENABLE = no  # Emulates clicks using speaker, who would want this?
+VARIABLE_TRACE = no     # Allows debugging variables
+API_SYSEX_ENABLE = no   # Allows OS to send signals.
```
Comment (fauxpark) [inline users/bbaserdem/rules.mk:36]:
```suggestion
```

File: users/bbaserdem/rules.mk [hunk 11]
```diff
@@ -1,21 +1,91 @@
...
+BLUETOOTH_ENABLE = no   # For bluetooth
```
Comment (fauxpark) [inline users/bbaserdem/rules.mk:37]:
```suggestion
```

File: users/bbaserdem/rules.mk [hunk 12]
```diff
@@ -1,21 +1,91 @@
...
+
+# These should be disabled in all boards
+BOOTMAGIC_ENABLE = no   # Access to EEPROM settings, not needed
+CONSOLE_ENABLE = no     # Allows console output with a command
+COMMAND_ENABLE = no     # Some bootmagic thing i dont use
+SLEEP_LED_ENABLE = no   # Breathes LED's when computer is asleep. Untested.
+MIDI_ENABLE = no        # Midi driver (untested)
+UNICODE_ENABLE = no 	# We use unicodemap, not unicode
+UCIS_ENABLE = no 	# We use unicodemap, not ucis
+FAUXCLICKY_ENABLE = no  # Emulates clicks using speaker, who would want this?
```
Comment (fauxpark) [inline users/bbaserdem/rules.mk:34]:
```suggestion
```

File: users/bbaserdem/rules.mk [hunk 13]
```diff
@@ -1,21 +1,91 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Common compilation flags
+EXTRAFLAGS += -flto     # Used to make code smaller
```
Comment (fauxpark) [inline users/bbaserdem/rules.mk:18]:
```suggestion
```

File: users/bbaserdem/bb-macro.c [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+        case TR_FLAG:
+            // Turkish flag
+            if (record->event.pressed) {
+                send_unicode_string("🇹🇷");
+            }
+            return false; break;
+        case BB_LENY:
+            // Lenny face: ( ͡° ͜ʖ ͡°)
+            if (record->event.pressed) {
+                send_unicode_hex_string("0028 0020 0361 00B0 0020 035C 0296 0020 0361 00B0 0029");
```
Comment (fauxpark) [inline users/bbaserdem/bb-macro.c:144]: Use `send_unicode_string()` please.

File: layouts/community/split_3x6_3/bbaserdem/README.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
+This file contains my [split 3x6 + 3](../../../default/split_3x6_3) board layouts
```
Comment (fauxpark) [inline layouts/community/split_3x6_3/bbaserdem/README.md:1]: ""

File: users/bbaserdem/README.md [hunk 16]
```diff
@@ -1,51 +1,131 @@
-# Overview
```
Comment (fauxpark) [inline users/bbaserdem/README.md:1]: ""

File: layouts/community/split_3x5_3/bbaserdem/README.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
+This file contains my [split 3x10 + 3](../../../default/split_3x10_3) board layouts
```
Comment (fauxpark) [inline layouts/community/split_3x5_3/bbaserdem/README.md:1]: ""

File: layouts/community/split_3x5_3/bbaserdem/README.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+Using a no 6th column crkbd (rev 2.1 choc) as a mobile keyboard.
+
+This version uses ProMicro as controllers; can't have OLED and RGB at the same time.
+
+## Build
+
+Used to flash Pro Micro explicitly using the following line;
+
+```
+sudo avrdude -p atmgea34u4 -P "$(ls /dev/ttyACM*)" -c avr109 -D -U flash:w:.build/<keeb>.hex
```
Comment (fauxpark) [inline layouts/community/split_3x5_3/bbaserdem/README.md:15]: `qmk flash` should be used here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: layouts/community/split_3x5_3/bbaserdem/README.md (hunk 18 above)
Comment (bbaserdem) [reply layouts/community/split_3x5_3/bbaserdem/README.md:15]: Added clarification here; it's a line i use to flash a hex file to pro micros. I have been having issues with qmk flash on this board; so I reference this line. Added clarification.

File: users/bbaserdem/bbaserdem.h [hunk 19]
```diff
@@ -1,268 +1,573 @@
...
+#define xxx1xxx KC_NO
+#define xxx2xxx KC_NO,  KC_NO
+#define xxx3xxx KC_NO,  KC_NO,  KC_NO
+#define xxx4xxx KC_NO,  KC_NO,  KC_NO,  KC_NO
+#define xxx5xxx KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO
+#define xxx6xxx KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO
+#define xxx1xxx KC_NO
+#define xxx3xxx KC_NO,  KC_NO,  KC_NO
+#define xxx5xxx KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO
+#define xxx6xxx KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO
```
Comment (noroadsleft) [inline users/bbaserdem/bbaserdem.h:196]: This block is redundant.

```suggestion
```

File: users/bbaserdem/readme.md [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+Pro Micro VCC      <-> Keyboard VCC
+Pro Micro GND      <-> Keyboard GND
+```
+I do have this on hand I believe; from massdrop's planck light firmware updater.
+3. Connect the flashed pro micro to my computer and run
+`avrdude -c avrisp -P "$(ls /dev/ttyACM*)" -p atmega32u4 -U flash:w:bootloader_atmega32u4_1.0.0.hex:i -U lfuse:w:0x5E:m -U hfuse:w:0xD9:m -U efuse:w:0xC3:m`
+
+The avrisp here refers to the firmware on the ISP flasher pro micro.
+The `atmega32u4` refers to the CPU used in the respective breakout boards.
+The `avrdude` command interacts with catalina bootloader.
```
Comment (noroadsleft) [inline users/bbaserdem/readme.md:101]: The bootloader's name is actually Caterina.

```suggestion
The `avrdude` command interacts with Caterina bootloader.
```

File: users/bbaserdem/readme.md [hunk 21]
```diff
@@ -0,0 +1,131 @@
...
+
+My userspace has a lot of shared code between different keyboards.
+These files are prefixed with `sbp-` to remove any naming clash.
+
+* [bb-audio](bb-audio.c): Code dealing with audio playback using onboard sound.
+* [bb-encoder](bb-encoder.c): Rotary encoder sutff.
+* [bb-macro](bb-macro.c): My custom keycodes; macros, tap dances, etc.
+* [bb-oled](bb-oled.c): Controls OLED displays. For higher memory; there is also [bb-oled-extra](bb-oled-extra.c).
+* [bb-rgb](bb-rgb.c): Controls per-key RGB LED matrix stuff, and layer indication. Uses `RGB_MATRIX`.
+* [bb-underglaw](bb-underglow.c): Controls RGB LED strip, and layer indication. Uses `RGBLIGHT`.
```
Comment (noroadsleft) [inline users/bbaserdem/readme.md:115]:
```suggestion
* [bb-underglow](bb-underglow.c): Controls RGB LED strip, and layer indication. Uses `RGBLIGHT`.
```

File: users/bbaserdem/readme.md [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+# Features
+
+My userspace has a lot of shared code between different keyboards.
+These files are prefixed with `sbp-` to remove any naming clash.
+
+* [bb-audio](bb-audio.c): Code dealing with audio playback using onboard sound.
+* [bb-encoder](bb-encoder.c): Rotary encoder sutff.
+* [bb-macro](bb-macro.c): My custom keycodes; macros, tap dances, etc.
+* [bb-oled](bb-oled.c): Controls OLED displays. For higher memory; there is also [bb-oled-extra](bb-oled-extra.c).
+* [bb-rgb](bb-rgb.c): Controls per-key RGB LED matrix stuff, and layer indication. Uses `RGB_MATRIX`.
```
Comment (noroadsleft) [inline users/bbaserdem/readme.md:114]: No mention of `bb-backlight`?

**Review by noroadsleft — COMMENTED** [review]
Compilation wise, everything seems to be in order here. I think the only thing truly needed is a rebase, because our copy of `keyboards/xiudi/xd75/keymaps/bbaserdem/readme.md` has been updated since your last pull request, so there's a merge conflict preventing us from merging this PR.

**Review by tzarc — APPROVED** [review]

**Pull request merged**